packages/common/src/utils/sagaHelpers.ts

```diff
@@ -138,3 +147,33 @@ export function* waitForRead() {
   yield* call(waitForReachability)
   yield* call(waitForAccount)
 }
+
+/**
+ * Yields the provided saga, wrapping it in an error
+ * boundary that restarts the saga and toasts if it fails.
+ * @param saga
+ */
+export function* sagaWithErrorHandler(saga: () => Generator<any, void, any>) {
+  while (true) {
+    try {
+      yield* call(saga)
+      break
+    } catch (e) {
+      yield* put(toast({ content: messages.somethingWrong }))
+      console.error(`Saga ${saga.name} failed, restarting...`, e)
```

Comment:
I _think_ this will result in two reports to sentry per error? Since sentry picks up `console.error()` and pipes it through automatically.
IIRC, most of the sagas that catch errors and reportToSentry don't do separate console statements since the call to `reportToSentry()` will spit out it's own console.error()
Would recommend making this warn/info.

Reply:
I think we should probably stop sentry from picking up console.errors tbh. It feels kinda chaotic to do so. wdyt? will change to warn here though

Reply:
I'm open to it. I like that it helps us find the errors we aren't catching. But it might be the case that Sentry catches exceptions and reports them independently of `console.error()` statements. If so, I vote we ditch the console reporting. I think it ends up being a lot of third-party code and things that aren't actually actionable.